server/models/groups/addBranch.js

```diff
@@ -14,70 +14,68 @@ function addBranch({ params, body, username }) {
             }
-            let processedUsernames = 0
-            let invalidUsernames = []
-            let alreadyMembers = []
-            results.forEach((member) => {
-              pool.query(
-                `INSERT INTO user_groups (username, group_id, is_group_moderator) VALUES(?,?,0)`,
-                [member.username, groupId],
-                (error, insertResults) => {
-                  processedUsernames++
-                  if (error || insertResults === undefined) {
-                    if (error.code === 'ER_DUP_ENTRY') {
-                      alreadyMembers.push(member.username)
-                    } else {
-                      invalidUsernames.push(member.username)
-                    }
-                  }
-                  if (processedUsernames === results.length) {
-                    pool.query(
-                      `UPDATE \`groups\` SET member_count = member_count + ? WHERE id=?`,
-                      [
-                        results.length -
-                          invalidUsernames.length -
-                          alreadyMembers.length,
-                        groupId,
-                      ],
-                      (error) => {
-                        if (error) {
-                          return reject(error)
-                        }
-                        return resolve({
-                          message: 'Branch added successfully!',
-                          invalidUsernames: invalidUsernames,
-                          alreadyMembers: alreadyMembers,
-                        })
-                      }
-                    )
-                  }
+            const { affectedRows } = results
+            if (!affectedRows) {
+              return connection.rollback(() => {
+                connection.release()
+                return reject(
+                  'Either you do not have moderator access of the group or there is no user with this branch'
+                )
+              })
+            }
+            connection.query(
+              `UPDATE \`groups\` SET member_count = (SELECT COUNT(username) FROM user_groups WHERE group_id=?) WHERE id=?`,
+              [groupId, groupId],
+              (error) => {
+                if (error) {
+                  return connection.rollback(() => {
+                    connection.release()
+                    return reject(error)
+                  })
                 }
-              )
-            })
+                connection.commit((error) => {
+                  if (error) {
+                    return connection.rollback(() => {
+                      connection.release()
+                      return reject(error)
+                    })
+                  }
+                  connection.release()
+                  return resolve({
+                    message: 'Branch added successfully!',
+                    branchMembersCount: affectedRows,
+                  })
+                })
+              }
+            )
```

Comment:
In API for for adding a branch to a group, used a transaction (2 queries) instead of iterating all usernames belonging to a  branch